src/posix/platform/settings.cpp

```diff
@@ -494,6 +500,12 @@ void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
     memset(aIeeeEui64, 0, sizeof(uint64_t));
 }
 
+// Stub implementation for testing
+bool IsSystemDryRun(void)
```

Comment:
I think it should cause duplicate definitions for `IsSystemDryRun`. 

Why it is not the case?

Reply:
Good question! This definition is added for the unit test of the settings module(https://github.com/openthread/openthread/blob/main/src/posix/platform/Makefile.am#L105):
```
test_settings_SOURCES                     = \
    settings.cpp                            \
    $(NULL)
``` 
In the unit test program, `system.cpp` isn't compiled together. And in the usual build, the definition is wrapped by `SELF_TEST`.